mautic/mautic
# Add  "If exist, not overwrite" to form to contact field mapping  tab

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/mautic-documentation/pull/279
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added new option to form fields **If exist, not overwrite**. 
This option don't allow overwrite lead field value If exists.

![image](https://user-images.githubusercontent.com/462477/70318715-fbc17080-1820-11ea-8d97-f3a81fb8b6df.png)



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Run the migration:`ddev exec bin/console d:m:e 20191206110334 --up`
2. Create form and set contact field mapping (firstname, lastname, email)
3. Try submit form your contact should be overwritten 
4. Try submit again, with same email but different firstname, lastname. Your contact should be overwritten
5. Now go to form, edit firstname field and enable new option **If exist, not overwrite**. 
6. Try submit again, with same email but different firstname, lastname. Your contact should be overwritten, but firstname shouldn't
7. Roll back the migration:`ddev exec bin/console d:m:e 20191206110334 --down`


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/8196"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
